server/container/guice/postgres-common/src/main/java/org/apache/james/modules/blobstore/BlobStoreConfiguration.java

```diff
@@ -0,0 +1,229 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.modules.blobstore;
+
+import java.io.FileNotFoundException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.james.blob.aes.CryptoConfig;
+import org.apache.james.modules.mailbox.ConfigurationComponent;
+import org.apache.james.server.blob.deduplication.StorageStrategy;
+import org.apache.james.server.core.filesystem.FileSystemImpl;
+import org.apache.james.utils.PropertiesProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.MoreObjects;
+
+public class BlobStoreConfiguration {
```

Comment:
Please do NOT duplicate this file. 

IMO we must reuse the one defined in `server/container/guice/distributed/src/main/java/org/apache/james/modules/blobstore/BlobStoreConfiguration.java`.

We then can validate later, EG in a `StartUpCheck` that the blob store cache is not activated for the postgres application.

Reply:
I found it hard to reuse. Because the configuration could be diverged. e.g in `server/container/guice/distributed/src/main/java/org/apache/james/modules/blobstore/BlobStoreConfiguration.java`.
```
       } catch (FileNotFoundException e) {
            LOGGER.warn("Could not find " + ConfigurationComponent.NAME + " configuration file, using cassandra blobstore as the default");
            return BlobStoreConfiguration.builder()
                .cassandra()
```

Put a configurable default blobStore choice (e.g. file/postgres) to the method?

The same thing for cache configuration. It is not the case for postgres-app now (unless we plan to implement blob cache with Postgres too?). Make the cache configuration optional? 

My feeling is that it is do-able to reuse, but it could be misleading too.

Reply:
Maybe a bit tricky to refactor. I agree it would be cleaner to have something like the SearchModuleChooser that we can reuse on any implem. But it requires some refactoring here.

Ticket?

Reply:
>? Maybe a bit tricky to refactor. I agree it would be cleaner to have something like the SearchModuleChooser that we can reuse on any implem. But it requires some refactoring here.

Let me try to handle this in this PR.

Reply:
fully reuse reuse `BlobStoreConfiguration` + `BlobStoreModulesChooser` now.